src/test/java/bio/terra/service/search/SearchTest.java

```diff
@@ -0,0 +1,105 @@
+package bio.terra.service.search;
+
+import bio.terra.common.category.Unit;
+import bio.terra.model.SearchIndexRequest;
+import bio.terra.service.resourcemanagement.google.GoogleProjectResource;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotTable;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.mockito.Mockito.when;
+
+@Category(Unit.class)
+public class SearchTest {
+    private static final String sqlQuery = "SELECT GENERATE_UUID() uuid, CURRENT_TIMESTAMP() as now FROM UNNEST" +
+        "(GENERATE_ARRAY(1, 3));";
+
+    @Mock
+    private BigQueryPdao bigQueryPdao;
+
+    @Mock
+    private RestHighLevelClient client;
+
+    private SearchService service;
+
+    private SearchIndexRequest searchIndexRequest;
+    private Snapshot snapshot;
+    private List<Map<String, Object>> values;
+
+    @Before
+    public void setup() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        service = new SearchService(bigQueryPdao, client);
+
+        searchIndexRequest = getSearchIndexRequest();
+        snapshot = getSnapshot();
+        values = getSnapshotTableData();
```

Comment:
We do this in a lot of our tests, but one thing to think about is that `@Before` is called before every test, and it's better to only put things in it that need to be reinitialized before every test. These objects appear to be static so they could be done in a `@BeforeClass`.

Reply:
I tried using a `@BeforeClass` , but I get the following error:

```
> No tests found for given includes: [bio.terra.service.search.SearchServiceTest.indexSnapshotTest](filter.includeTestsMatching)
```

Reply:
The `BeforeClass` method needs to be public and static, maybe that was the issue? https://junit.org/junit4/javadoc/4.13/org/junit/BeforeClass.html

Reply:
It seems to be interacting poorly with Mockito. If I set the method to static, then all the constructor member variables need to also be static, but this causes an error with the `@Mock` annotation.

Reply:
Ah well, never mind then. Thanks for trying it out.